### PR TITLE
SINGA-474 Clip operator

### PR DESCRIPTION
**Pull request by ShichengChen**
SINGA-474 Clip operator for autograd and ONNX

### CODE REVIEW

**Comment by nudles** [comment]
the same code appears in both #503 and #502 .
you can add feature A, commit and send PR A; then add feature B, commit and send PR B; 
you can also skip PR A and just send PR B.
but when you have feature A, B, ...; then you'd better split this big PR into multiple small PRs.

**Comment by ShichengChen** [comment]
> the same code appears in both #503 and #502 .
> you can add feature A, commit and send PR A; then add feature B, commit and send PR B;
> you can also skip PR A and just send PR B.
> but when you have feature A, B, ...; then you'd better split this big PR into multiple small PRs.

I just remove useless code and split them into two PR

File: python/singa/autograd.py [hunk 1]
```diff
@@ -357,6 +359,46 @@ def backward(self, dy):
...
+        mask3 = singa.GEFloat(x, self.min)
+        maskm = singa.__mul__(mask2,mask3)
+
+        if training:
+            self.mask = maskm
+
+        return singa.__add__(singa.__add__(singa.__mul__(maskm,x),mask00),mask11)
+
+    def backward(self, dy):
+        """
```
Comment (nudles) [inline python/singa/autograd.py]: the comments are misleading..

Comment (ShichengChen) [reply python/singa/autograd.py]: removed misleading comments

**Pull request merged**